File: twidere/src/main/kotlin/org/mariotaku/twidere/adapter/iface/IItemCountsAdapter.java
package org.mariotaku.twidere.adapter.iface;

/**
 * Created by mariotaku on 16/8/19.
 */
interface IItemCountsAdapter {

    val itemCounts: IntArray

    fun getItemCountIndex(position: Int): Int {
        var sum: Int = 0
        itemCounts.forEachIndexed { idx, count ->
            sum += count
            if (position < sum) {
                return idx
            }
        }
        return -1
    }

    fun getItemStartPosition(index: Int): Int {
        return itemCounts.slice(0 until index).sum()
    }

}

File: twidere/src/main/kotlin/org/mariotaku/twidere/adapter/iface/IItemCountsAdapter.kt
package org.mariotaku.twidere.adapter.iface;

import org.mariotaku.twidere.model.ItemCounts

/**
 * Created by mariotaku on 16/8/19.
 */
interface IItemCountsAdapter {

    val itemCounts: ItemCounts

    fun getItemCountIndex(position: Int): Int {
        return itemCounts.getItemCountIndex(position)
    }

    fun getItemStartPosition(index: Int): Int {
        return itemCounts.getItemStartPosition(index)
    }

}